executor/merge_join.go

```diff
@@ -340,6 +342,22 @@ func (e *MergeJoinExec) joinToChunk(ctx context.Context, chk *chunk.Chunk) (hasM
 	}
 }
 
+func (e *MergeJoinExec) compare(outerRow, innerRow chunk.Row) (int, error) {
+	outerJoinKeys := e.outerTable.keys
+	innerJoinKeys := e.innerTable.joinKeys
+	for i := range outerJoinKeys {
+		cmp, _, err := e.compareFuncs[i](e.ctx, outerJoinKeys[i], innerJoinKeys[i], outerRow, innerRow)
```

Comment:
We do not check the null-value?

Reply:
yes, I've refactored the `expression.CompareFunc`s to compare NULL values, compare result are stored in the first return value, thus the second return value only indicates whether there is a NULL value.